File: 0001_0100/0037/0037.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, k = f.readline().split()
    n = int(n)
    k = round(float(k) * 1000)
    while n:
        a, b, c, d = map(int, f.readline().split())
        if 1000000 * (c * c + d * d) > k * k * (a * a + b * b):
            break
        n -= 1
    q.write("No" if n else "Yes")

File: 0001_0100/0037/0037.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	var k float64
	var a, b, c, d int64
	fmt.Fscan(f, &n, &k)
	j := int64(1000 * k)
	r := "Yes"
	for ; n > 0; n-- {
		fmt.Fscan(f, &a, &b, &c, &d)
		if 1000000*(c*c+d*d) > j*j*(a*a+b*b) {
			r = "No"
			break
		}
	}
	fmt.Fprint(q, r)
}
